### PR TITLE
Code structure refactor + erc20

### PR DESCRIPTION
**Pull request by zie1ony**

### CODE REVIEW

File: dao-erc20/tests/test_erc20.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        3,
+        Approval {
+            owner,
+            spender,
+            value: U256::zero(),
+        },
+    );
+}
+
+fn main() {}
```
Comment (kpob) [inline dao-erc20/tests/test_erc20.rs:155]:
```suggestion
```

File: dao-utils/src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,11 @@
 extern crate alloc;
 
+// Reexport of casper-contract crate.
+pub use casper_contract;
```
Comment (kpob) [inline dao-utils/src/lib.rs:4]: What's the benefit?

File: dao-utils/src/casper_env.rs [hunk 3]
```diff
@@ -122,6 +122,6 @@ pub fn get_block_time() -> u64 {
     u64::from(runtime::get_blocktime())
 }
 
-pub fn revert<T: Into<ApiError>>(error: T) {
-    runtime::revert(error)
+pub fn revert<T: Into<ApiError>>(error: T) -> ! {
```
Comment (kpob) [inline dao-utils/src/casper_env.rs:125]: Does it change the function's behavior?

File: Makefile [hunk 4]
```diff
@@ -27,7 +41,14 @@ lint: clippy
...
-	cargo doc --features test-support --no-deps --open
\ No newline at end of file
+	cargo doc --features test-support --no-deps --open
+
+github-test: build-proxy-getter build-dao-contracts build-erc20
+	mkdir -p dao-contracts/wasm
+	mkdir -p dao-erc20/wasm
+	cp $(OUTPUT_DIR)/*.wasm dao-contracts/wasm
+	cp $(OUTPUT_DIR)/*.wasm dao-erc20/wasm
+	$(CARGO_JUST_TEST) -p casper-dao-contracts --test test-reputation --test test-variable-repository
```
[the remark this answers is not in the document]
Comment (zie1ony) [reply Makefile:53]: I think it's a small price for a cleaner test output.

File: dao-contracts/bin/reputation_contract.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#![no_main]
 use casper_dao_contracts::{ReputationContract, ReputationContractInterface};
-
 casper_dao_contracts::reputation_contract!();
+
+fn main() {}
```
[the remark this answers is not in the document]
Comment (zie1ony) [reply dao-contracts/bin/reputation_contract.rs:4]: Have you tried running tests without it?

File: dao-utils/src/lib.rs (hunk 2 above)
Comment (zie1ony) [reply dao-utils/src/lib.rs:4]: In general I would like no to use `casper_contract` directly and use our `casper_env`. It also makes features and dependencies setting a bit cleaner.

File: dao-utils/src/casper_env.rs (hunk 3 above)
Comment (zie1ony) [reply dao-utils/src/casper_env.rs:125]: `runtime::revert` is like panic, it ends the execution. Here I wrapped it with another function and want to have the same behavior, so I need `-> !`.

File: dao-contracts/bin/reputation_contract.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kpob) [reply dao-contracts/bin/reputation_contract.rs:4]: Ok, it doesn't work. But... it used to work before the refactor. :confused:

File: dao-erc20/tests/test_erc20.rs (hunk 1 above)
Comment (kpob) [reply dao-erc20/tests/test_erc20.rs:155]: Btw. works perfectly fine without `main()` it tests

**Review by kpob — APPROVED** [review]

**Pull request merged**